napari/settings/_plugins.py

```diff
@@ -17,14 +17,27 @@ class PluginHookOption(TypedDict):
 
 class PluginsSettings(EventedSettings):
     use_npe2_adaptor: bool = Field(
-        False,
+        True,
         title=trans._('Use npe2 adaptor'),
         description=trans._(
             "Use npe2-adaptor for first generation plugins.\nWhen an npe1 plugin is found, this option will\nimport its contributions and create/cache\na 'shim' npe2 manifest that allows it to be treated\nlike an npe2 plugin (with delayed imports, etc...)",
         ),
         requires_restart=True,
     )
-
+    only_new_shimmed_plugins_warning: bool = Field(
+        False,
+        title=trans._('Only warn for new adapted plugins'),
+        description=trans._(
+            'Only warn about newly installed adapted plugins. Leave unchecked to receive warning with each startup.'
```

Comment:
I would simplify the naming a bit.

```suggestion
    new_plugins_warning_only: bool = Field(
        False,
        title=trans._('Warn only for new plugins'),
        description=trans._(
            'Warn only about newly installed adapted plugins. Leave unchecked to receive warning with each startup.'
```

Reply:
Hmm I actually think it's important that the word `adapted` be in there somewhere, because otherwise it's not clear why we're warning about the new plugins at all (even though it's under the `Use npe2 adaptor` setting, I'd like it to be more explicit). How about `new_adapted_plugin_warning` for the settings title and `Warn for new adapted plugins only` for the title text? I know it's not very short...

Reply:
I like the suggestion. I would remove new from the variable name of the setting. The title text is fine with or without new.

Reply:
In general, the problem with using new in a variable name is it's new relative to what. If you use `adapted_plugin_warning` it is easier to change the behavior over time without having to change the interface. Not a big deal though, it's just a suggestion. And I'm going to approve since it seems good to move forward with either.